lib/mistral_ai_summary_example/mistralai_summary_page.dart

```diff
@@ -51,84 +49,159 @@ class _MistralAISummaryPageState extends State<MistralAISummaryPage> {
-                        child: const Text('Summarize sample tex'),
-                      ),
-                      const SizedBox(height: 20),
-                      TextButton(
-                        onPressed: () => showDialog<String>(
-                          context: context,
-                          builder: (context) => const SampleTextDialogButton(),
-                        ),
-                        child: const Text('See sample text'),
-                      ),
-                    ],
+                  const SizedBox(height: 20),
+                  SummaryResultText(
+                    summaryResult: summaryResult,
+                    isLoading: summaryInProgress,
                   ),
                 ],
               ),
-              const SizedBox(height: 20),
-              const Text('Summarized Text:'),
-              const SizedBox(height: 10),
-              SelectableText(summaryResult),
-            ],
+            ),
           ),
+        );
+      },
+    );
+  }
+}
+
+class SummaryActions extends StatelessWidget {
+  const SummaryActions({
+    required this.onSummarize,
+    required this.summaryInputController,
+    super.key,
+  });
+
+  final VoidCallback onSummarize;
+  final TextEditingController summaryInputController;
+
+  @override
+  Widget build(BuildContext context) {
+    return Wrap(
+      alignment: WrapAlignment.center,
+      runSpacing: 16,
+      children: [
+        ElevatedButton(
+          onPressed: onSummarize,
+          child: const Text('Summarize'),
+        ),
+        const SizedBox(width: 16),
+        SampleTextPasteButton(
+          summaryInputController: summaryInputController,
+        ),
+      ],
+    );
+  }
+}
+
+class SummaryResultText extends StatelessWidget {
+  const SummaryResultText({
+    required this.summaryResult,
+    required this.isLoading,
+    super.key,
+  });
+
+  final String summaryResult;
+  final bool isLoading;
+
+  @override
+  Widget build(BuildContext context) {
+    if (isLoading) {
+      return const Center(child: CircularProgressIndicator());
+    }
+    if (summaryResult.isEmpty) {
+      return const SizedBox.shrink();
+    }
+    return Column(
+      children: [
+        Text(
+          'Summary:',
+          style: Theme.of(context).textTheme.bodyLarge,
+        ),
+        const SizedBox(height: 10),
+        SelectableText(summaryResult),
+      ],
+    );
+  }
+}
+
+class SummaryTextInput extends StatelessWidget {
+  const SummaryTextInput({
+    required this.summaryInputController,
+    super.key,
+  });
+
+  final TextEditingController summaryInputController;
+
+  @override
+  Widget build(BuildContext context) {
+    return TextField(
+      controller: summaryInputController,
+      decoration: const InputDecoration(
+        border: OutlineInputBorder(),
+        labelText: 'Enter text to summarize',
+      ),
+      minLines: 1,
+      maxLines: 10,
+      onChanged: (value) => summaryInputController.text = value,
+    );
```

Comment:
I would add `autofocus` on this one. It will make the keyboard appear automatically on mobile.

Reply:
Resolved in [a26f9a3](https://github.com/nomtek/flutter_ai_examples/pull/30/commits/a26f9a3e7590bb5457aeda4e01ba02ba18f880f8)